lib/go/test/flow_epoch_test.go

```diff
@@ -1516,3 +1517,203 @@ func TestEpochReset(t *testing.T) {
+
+		// seed is not manually set when recovering the epoch, it is randomly generated
+		metadataFields := getEpochMetadata(t, b, env, cadence.NewUInt64(startEpochCounter+1))
+		seed := strings.ReplaceAll(metadataFields["seed"].String(), `"`, "")
+		expectedMetadata := EpochMetadata{
+			counter:               startEpochCounter + 1,
+			seed:                  seed,
+			startView:             startView,
+			endView:               endView,
+			stakingEndView:        stakingEndView,
+			totalRewards:          "0.0",
+			rewardsBreakdownArray: 0,
+			rewardsPaid:           false,
+			collectorClusters:     nil,
+			clusterQCs:            nil,
+			dkgKeys:               dkgPubKeys,
+		}
+		verifyEpochMetadata(t, b, env, expectedMetadata)
+
+		result := executeScriptAndCheck(t, b, templates.GenerateGetDKGEnabledScript(env), nil)
+		assert.Equal(t, cadence.NewBool(false), result)
+
+		result = executeScriptAndCheck(t, b, templates.GenerateGetQCEnabledScript(env), nil)
+		assert.Equal(t, cadence.NewBool(false), result)
+
+		expectedRecoverEvent := EpochRecover{
+			counter:                 startEpochCounter + 1,
+			nodeInfoLength:          len(nodeIDs),
+			firstView:               startView,
+			finalView:               endView,
+			collectorClusters:       collectorClusters,
+			randomSource:            seed,
+			dkgPhase1FinalView:      startView + numStakingViews + numDKGViews - 1,
+			dkgPhase2FinalView:      startView + numStakingViews + (2 * numDKGViews) - 1,
+			dkgPhase3FinalView:      startView + numStakingViews + (3 * numDKGViews) - 1,
+			targetDuration:          targetDuration,
+			targetEndTime:           targetEndTime,
+			clusterQCVoteDataLength: 0,
+			dkgPubKeys:              dkgPubKeys,
+		}
+		verifyEpochRecover(t, adapter, idTableAddress, expectedRecoverEvent)
+
+		// If epoch recover was rejected by the protocol state we can update the configuration
+		// for the epoch without creating a new one.
+		tx = createTxWithTemplateAndAuthorizer(b, templates.GenerateRecoverEpochScript(env), idTableAddress)
+
+		// change startView and endView
+		newStartView := startView + 1
+		args[0] = CadenceUInt64(newStartView)
+
+		newEndView := endView + 1
+		args[2] = CadenceUInt64(newEndView)
+
+		// avoid initializing a new epoch
+		args[9] = cadence.NewBool(false)
+		for _, arg := range args {
+			tx.AddArgument(arg)
+		}
+
+		signAndSubmit(
+			t, b, tx,
+			[]flow.Address{idTableAddress},
+			[]sdkcrypto.Signer{IDTableSigner},
+			false,
+		)
+
+		expectedRecoverEvent.firstView = newStartView
+		expectedRecoverEvent.finalView = newEndView
+		expectedRecoverEvent.dkgPhase1FinalView = newStartView + numStakingViews + numDKGViews - 1
+		expectedRecoverEvent.dkgPhase2FinalView = newStartView + numStakingViews + (2 * numDKGViews) - 1
+		expectedRecoverEvent.dkgPhase3FinalView = newStartView + numStakingViews + (3 * numDKGViews) - 1
+		verifyEpochRecover(t, adapter, idTableAddress, expectedRecoverEvent)
```

Comment:
Can we also validate: 
- `curEpochCounter = startEpochCounter+1`
- check `epochMetadata` in addition to the EpochRecover event

Reply:
https://github.com/onflow/flow-core-contracts/pull/420/commits/614f489fb0448a5278b4eec15cee7742701370ae